### PR TITLE
python311Packages.qcodes: 0.44.1 -> 0.45.0

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/QCoDeS/Qcodes/releases/tag/v0.45.0

cc @evilmav 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320532` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python312Packages.qcodes</li>
    <li>python312Packages.qcodes-contrib-drivers</li>
    <li>python312Packages.qcodes-contrib-drivers.dist</li>
    <li>python312Packages.qcodes.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320532` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python312Packages.qcodes</li>
    <li>python312Packages.qcodes-contrib-drivers</li>
    <li>python312Packages.qcodes-contrib-drivers.dist</li>
    <li>python312Packages.qcodes.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320532` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python312Packages.qcodes-contrib-drivers</li>
    <li>python312Packages.qcodes-contrib-drivers.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python312Packages.qcodes</li>
    <li>python312Packages.qcodes.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 320532` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.qcodes</li>
    <li>python311Packages.qcodes-contrib-drivers</li>
    <li>python311Packages.qcodes-contrib-drivers.dist</li>
    <li>python311Packages.qcodes.dist</li>
    <li>python312Packages.qcodes</li>
    <li>python312Packages.qcodes-contrib-drivers</li>
    <li>python312Packages.qcodes-contrib-drivers.dist</li>
    <li>python312Packages.qcodes.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/qcodes/default.nix [hunk 1]
```diff
@@ -79,14 +77,12 @@ buildPythonPackage rec {
     jsonschema
     matplotlib
     numpy
-    opencensus
-    opencensus-ext-azure
```
Comment (natsukium) [inline pkgs/development/python-modules/qcodes/default.nix:83]: Could you add these dependencies to `optional-dependencies`?
https://github.com/microsoft/Qcodes/compare/refs/tags/v0.44.1...v0.45.0#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711

Comment (GaetanLepage) [reply pkgs/development/python-modules/qcodes/default.nix:83]: Done (along other optional dependencies).

File: pkgs/development/python-modules/qcodes/default.nix [hunk 2]
```diff
@@ -97,6 +109,34 @@ buildPythonPackage rec {
     xarray
   ] ++ lib.optionals (pythonOlder "3.10") [ importlib-metadata ];
 
+  passthru.optional-dependencies = {
```
Comment (natsukium) [inline pkgs/development/python-modules/qcodes/default.nix:112]:
```suggestion
  optional-dependencies = {
```
We no longer need `passthru`.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**